File: multipaz-compose/src/commonMain/kotlin/org/multipaz/compose/items/FloatingItemCenteredText.kt
package org.multipaz.compose.items

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.style.TextAlign

/**
 * An item showing centered italicized text in the secondary/muted color.
 *
 * This is intended to be used in an [FloatingItemList] for showing that it's empty, e.g. [text] would
 * be "Gizmos will appear here, use + to add one".
 *
 * @param text the text to show.
 * @param modifier a [androidx.compose.ui.Modifier].
 * @param showChevron whether to show a right chevron icon on the right side.
 */
@Composable
fun FloatingItemCenteredText(
    text: AnnotatedString,
    modifier: Modifier = Modifier,
    showChevron: Boolean = false,
) {
    FloatingItemContainer(
        modifier = modifier,
        showChevron = showChevron,
    ) {
        Column(
            modifier = Modifier.fillMaxWidth(),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text(
                text = text,
                textAlign = TextAlign.Center,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                fontStyle = FontStyle.Italic
            )
        }
    }
}

/**
 * An item showing centered italicized text in the secondary/muted color.
 *
 * This is intended to be used in an [FloatingItemList] for showing that it's empty, e.g. [text] would
 * be "Gizmos will appear here, use + to add one".
 *
 * @param text the text to show.
 * @param modifier a [Modifier].
 * @param showChevron whether to show a right chevron icon on the right side.
 */
@Composable
fun FloatingItemCenteredText(
    text: String,
    modifier: Modifier = Modifier,
    showChevron: Boolean = false,
) {
    FloatingItemCenteredText(
        text = AnnotatedString(text),
        modifier = modifier,
        showChevron = showChevron,
    )
}

File: multipaz-swiftui/src/iosMain/swift/FloatingItemCenteredText.swift
import SwiftUI

/// A list item view that displays centered, italicized text in a secondary color.
///
/// This component is commonly used for informational footers, empty states, or subtle hints.
public struct FloatingItemCenteredText: View {

    /// The text to display, formatted as an `AttributedString`.
    public var text: AttributedString

    /// Whether to show a right chevron icon on the right side.
    public var showChevron: Bool

    /// Creates a new centered text item using an `AttributedString`.
    ///
    /// - Parameters:
    ///   - text: The formatted string to display centered and italicized.
    ///   - showChevron: Whether to show a right chevron icon on the right side. Defaults to `false`.
    public init(text: AttributedString, showChevron: Bool = false) {
        self.text = text
        self.showChevron = showChevron
    }

    /// Creates a new centered text item using a standard `String`.
    ///
    /// - Parameters:
    ///   - text: The string to display centered and italicized.
    ///   - showChevron: Whether to show a right chevron icon on the right side. Defaults to `false`.
    public init(text: String, showChevron: Bool = false) {
        self.init(text: AttributedString(text), showChevron: showChevron)
    }

    public var body: some View {
        FloatingItemContainer(showChevron: showChevron) {
            Text(text)
                .font(.system(size: 15))
                .multilineTextAlignment(.center)
                .foregroundColor(.secondary)
                .italic()
                .frame(maxWidth: .infinity, alignment: .center)
        }
    }
}
